src/hssm/plotting/posterior_predictive.py

```diff
@@ -0,0 +1,199 @@
+"""Plotting functionalities for HSSM."""
+
+import logging
+
+import arviz as az
+import matplotlib as mpl
+
+# import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+import seaborn as sns
+
+from .utils import _get_plotting_df
+
+_logger = logging.getLogger("hssm")
+
+
+def plot_posterior_predictive(
+    model,
+    idata: az.InferenceData | None = None,
+    data: pd.DataFrame | None = None,
+    n_samples: int | float | None = 20,
+    row: str | None = None,
```

Comment:
Yep. I am not sure about the usefulness of another dimension tbh

Reply:
Consider the case where a dataset actually has three categorical variables that specify groupings. 

In this case we could e.g. 

1. Take one of them, and and make a plot for each of it's values --> subplots across the other two

This would generalize to 4,5 etc. groupings

2. Warn users that they should subset their data according to the leading groups, and only pass data that has (up to) two relevant group dimensions left.

Reply:
As another way to think about it:

We could organize the computation by sub graph. 

1. Kernel:
You subset the dataframe, run the corresponding posterior predictive, use it to fill in one axis object. 

2. Scaffolding code:
Organizes the returned axes objects into a grid (potentially splitting them across plots etc.)

This way of thinking also allows (because prob more efficient from a simulation overhead viewpoint):
1. Run big posterior predictive simulation up front
2. subset posterior predictives and pass them to kernels
3. build larger plot

I acknowledge that this way of doing things might make seaborn a little less appealing, however retains flexibility on our end. 

What I mean is that we can think of `plots` as kernels, and retain and reuse the `scaffolding wrapper`.

So for new plots we just think about building plotting kernels.

Reply:
I totally see how it is possible to implement this. I am just not sure if this is a good visualization practice, once we have these high-dimensional visualizations. There is already a lot of information in the two dimensional case. So I guess it's more about me not wanting to encourage practice like this lol. However, if you have seen enough of a demand for higher dimensional charts, I am happy to implement this

Reply:
Take it on faith for now that people might want that :D.
I will motivate this a bit more during our next meeting.

Reply:
Then maybe let's allow for another dimension called `group`? Anything more than that, I feel like we should just make it possible for the users to use for loops themselves with the 2D version of the plotting function

Reply:
To make it general, just allow for group to be a list of covariates?
Then turn the combination of values into a top-level title for the respective 2D plots?

And in principle users can also go and loop over 2D plots themselves by passing the correct data-subsets as of now right?

Reply:
Okay @AlexanderFengler this is now added. The `groups` and `groups_order` arguments now supports arbitrary grouping dimensions.

> And in principle users can also go and loop over 2D plots themselves by passing the correct data-subsets as of now right?

Yep. This should be used in conjunction with `_get_plotting_df`, but now that the main function supports arbitrary grouping, this should not be necessary (though still possible)

Reply:
Great!